### PR TITLE
Add Dockerfile for easier building and testing.

### PR DESCRIPTION
**Pull request by tkaemming**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM postgres:9.6-alpine
```
Comment (tkaemming) [inline Dockerfile]: I know we looked to swap over to Ubuntu, we can either make that change here or as a follow up. Just trying to break up my branch into more independent changes.

Comment (tkaemming) [reply Dockerfile]: Moved this with b409583 — the regression tests weren't working correctly with the Alpine image anyway (it has a different `diff` implementation) and I want to get those working before applying some of the other changes from my other branch.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+FROM postgres:9.6
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+        build-essential \
+        git \
+        postgresql-server-dev-9.6 && \
+    rm -rf /var/lib/apt/lists/*
```
Comment (tkaemming) [inline Dockerfile]: This is going to still leave around a bunch of stuff we don't need in the final image — eventually we should do the `make install` as part of this `RUN` command and `apt-get purge` everything at the end.

**Comment by tkaemming** [comment]
Tests can be run in the container with:

```bash
docker run -d --name wal2json --rm $(docker build -q .) -c wal_level=logical -c max_replication_slots=1
docker exec wal2json bash -c 'cd wal2json && PGUSER=postgres make installcheck'
```

(This obviously isn't particular ergonomic and can/should be improved.)

**Comment by tkaemming** [comment]
OK, so after taking with @mattrobenolt, I'm planning on making some updates:

- Moving back to Alpine (sorry @JTCunning.) People already run Alpine in development, and using an Ubuntu base image here will probably mess up their indexes or something. If we want to build the binary for Ubuntu in Docker (which it seems like to most of us seems questionable in the first place), we need to do this via a separate Dockerfile.
- Moving to [multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/). This way we can have the build image with all of the toolchain requirements, as well as a slimmed image for people to run in local development that only adds the shared library and the configuration changes. This also allows us to build this in development without having to reinstall the toolchain each time.
- Running the tests as part of the image build process since they require parts of the build toolchain. The only thing CI will do is try to build the image. If it succeeds, the tests pass and we're green; if it fails, the tests fail and we're red.

**Comment by JTCunning** [comment]
I'm just going to resort on building the library on the host with config-mgmt. Don't want to get in anyone else's way and also don't want to blow up PG because of dependency mismatches.

**Comment by tkaemming** [comment]
This should be good to go, unless there are changes that need to be made to the Dockerfile itself.

Decided to not run the tests as part of the build, since that relies on running `initdb` and a bunch of stuff that doesn't seem like it make sense to get into here. We can keep the tests as a separate script that automates starting the `build` image target, waiting for it to be ready, running the tests in the container, reporting results, and shutting it down.

Speaking of tests, the tests currently don't work in Alpine because the way the random number generator is seeded differs between glibc and musl. I'll fix the tests and get this wired up to CI, but going to wait until @fpacifici makes some broader changes to the test results first so we don't end up having to deal with as many merge conflicts down the line.

**Pull request merged**